waspc/data/Generator/templates/server/src/core/auth.js

```diff
@@ -57,9 +68,12 @@ export async function getUserFromToken(token) {
   // password field from the object here, we must to do the same there).
   // Ideally, these two things would live in the same place:
   // https://github.com/wasp-lang/wasp/issues/965
-  const { password, ...userView } = user
-
-  return userView
+  let sanitizedUser = { ...user }
+  sanitizedUser.{= authFieldOnUserEntityName =}.{= identitiesFieldOnAuthEntityName =} = sanitizedUser.{= authFieldOnUserEntityName =}.{= identitiesFieldOnAuthEntityName =}.map(identity => {
+    identity.providerData = deserializeProviderData(identity.providerData, { shouldRemovePasswordField: true })
```

Comment:
So deserialization does also sanitization? We have sanitizeAndSerialize function, why don't we also call this function deserializeAndSanitize then? if it is not appropriate we don't have to, but wanted to ask.

Reply:
Yep, that's good suggestion 👍

Reply:
I think I saw this one being done? If so pls throw a :rocket: .